src/Form/UpdateForm.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\workspace\Form;
+
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\CloseModalDialogCommand;
+use Drupal\Core\Ajax\PrependCommand;
+use Drupal\Core\Form\ConfirmFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\replication\Entity\ReplicationLogInterface;
+
+class UpdateForm extends ConfirmFormBase {
+
+  /**
+   * Get the current active workspace's pointer.
+   */
+  protected function getActive() {
+    $workspace = \Drupal::service('workspace.manager')->getActiveWorkspace();
+    $pointers = \Drupal::service('entity_type.manager')
+      ->getStorage('workspace_pointer')
+      ->loadByProperties(['workspace_pointer' => $workspace->id()]);
+    return reset($pointers);
+  }
+
+  /**
+   * Returns the upstream for the given workspace.
+   *
+   * @return \Drupal\multiversion\Entity\WorkspaceInterface
+   */
+  protected function getUpstream() {
+    $workspace = \Drupal::service('workspace.manager')->getActiveWorkspace();
+    if (isset($workspace->upstream)) {
+      return $workspace->upstream->entity;
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildForm($form, $form_state);
+    $form['actions']['submit']['#ajax'] = [
+      'callback' => [$this, 'update'],
+      'event' => 'mousedown',
+      'prevent' => 'click',
+      'progress' => [
+        'type' => 'throbber',
+        'message' => 'Updating',
+      ],
+    ];
+
+    if (!$this->getUpstream()) {
+      unset($form['actions']['submit']);
+    }
+    unset($form['actions']['cancel']);
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuestion() {
+    return $this->t('Update @workspace', ['@workspace' => $this->getActive()->label()]);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCancelUrl() {
+    return new Url('system.admin');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'workspace_update_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    try {
+      $response = \Drupal::service('workspace.replicator_manager')->update(
+        $this->getUpstream(),
+        $this->getActive()
+      );
+
+      if (($response instanceof ReplicationLogInterface) && $response->get('ok')) {
+        drupal_set_message('Successful update.');
```

Comment:
Let's provide a more useful message. Something like `The current workspace [name] has been updated with content from the upstream workspace [name].`